tgui/packages/tgui/interfaces/NtosMessenger/index.tsx

```diff
@@ -0,0 +1,339 @@
+                }
+                onClick={() => act('PDA_toggleSendingAndReceiving')}
+              />
+              <Button
+                icon="bell"
+                content="Set Ringtone"
+                onClick={() => act('PDA_ringSet')}
+              />
+              <Button
+                icon="sort"
+                content={`Sort by: ${sort_by_job ? 'Job' : 'Name'}`}
+                onClick={() => act('PDA_changeSortStyle')}
+              />
+              {!!virus_attach && (
+                <Button
+                  icon="bug"
+                  color="bad"
+                  content={`Attach Virus: ${sending_virus ? 'Yes' : 'No'}`}
+                  onClick={() => act('PDA_toggleVirus')}
+                />
+              )}
+            </Box>
+          </Stack>
+          <Divider hidden />
+          <Stack justify="space-between">
+            <Box m={0.5}>
+              <Icon name="magnifying-glass" mr={1} />
+              Search For User
+            </Box>
+            <Input
+              width="220px"
+              placeholder="Search by name or job..."
+              value={searchUser}
+              onInput={(_e: any, value: string) => setSearchUser(value)}
+            />
+          </Stack>
+        </Section>
+      </Stack.Item>
+      {filteredChatButtons.length > 0 && (
+        <Stack.Item>
+          <Stack fill vertical>
+            <Stack vertical>
+              <Section fill>
+                <Icon name="comments" mr={1} />
+                Previous Messages
+              </Section>
+            </Stack>
+            <Section maxHeight={20} scrollable mt={1}>
+              <Stack vertical>{filteredChatButtons}</Stack>
+            </Section>
+          </Stack>
+        </Stack.Item>
+      )}
+      <Stack.Item grow>
+        <Stack vertical fill>
+          <Section>
+            <Stack>
+              <Box m={0.5}>
+                <Icon name="address-card" mr={1} />
+                Detected Messengers
+              </Box>
+            </Stack>
+          </Section>
+          <Section fill scrollable>
+            <Stack vertical pb={1} fill>
+              {messengerButtons.length === 0 && (
+                <Stack align="center" justify="center" fill pl={4}>
+                  <Icon color="gray" name="user-slash" size={2} />
+                  <Stack.Item fontSize={1.5} ml={3}>
+                    No users found.
+                  </Stack.Item>
+                </Stack>
+              )}
+              {messengerButtons}
+            </Stack>
+          </Section>
+        </Stack>
+      </Stack.Item>
+      {!!can_spam && (
+        <Stack.Item>
+          <SendToAllSection />
+        </Stack.Item>
+      )}
+      {noId && <NoIDDimmer />}
+    </Stack>
+  );
+};
+
+type ChatButtonProps = {
+  name: string;
+  unreads: number;
+  chatRef: string;
+};
+
+const ChatButton: SFC<ChatButtonProps> = (props: ChatButtonProps, ctx: any) => {
```

Comment:
this is fine but just for the record you're effectively doing the same thing twice. i think typing components is kind of silly (what is it checking? that it returns a jsx element?). so either `SFC<Proptype>` or `props: Proptype` is valid, but both is unneeded

Reply:
removed all SFC declarations that aren't really needed, but I'll keep the one around for the `ChatDivider` component around because `(props: { mt: number })` looks ugly imo

Reply:
yeah, not a huge fan of inlining types either. others have arguments for that, it supposedly helps with the info given on hover.